### DISCUSSION TITLE
led-strip

### DISCUSSION THREAD

**Discussion by SergPiven** [UncleRus/esp-idf-lib#307] [Q&A] [answered]
Another question, I want to pour your firmware LED strip, to me to practice. Help pls!

**Comment by UncleRus** [comment, marked as answer]
> pour your firmware

"Upload" may be?

This is an ESP32 (ESP32 only! no ESP32-Cx/ESP32-Sx) project using led_strip (and some other components):
https://github.com/UncleRus/EvLamp

And here's a simpler example:
https://github.com/UncleRus/esp-idf-lib/tree/master/examples/led_effects

If it is not clear how to build and upload these projects to esp32, then you need to start by studying the ESP-IDF documentation:
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/index.html

Very roughly:

- Download ESP-IDF
- Run the install.sh script (or install.bat if you have Windows)
- Wait for the script to download toolchains
- Carefully read the result of the script in the console and follow the instructions
- Then you can build and upload firmware. To do this, you need to use the `idf.py` utility from the ESP-IDF

In short:
`idf.py menuconfig` - configure project before building
`idf.py build` - completely build the project (compile, link, convert to firmware)
`idf.py flash` - upload firmware to the microcontroller connected via serial port
`idf.py monitor` - open serial debug terminal and reset microcontroller
Arguments can be combined, for example:
`idf.py build flash monitor` - build firmware, upload it and show its logs

**Reply by UncleRus** [reply]
And don't hesitate to explore official Espressif forum: https://esp32.com/
